### ISSUE TITLE
New history refresh prevents push/tab notification of job completions

### ISSUE THREAD

**Issue by astrovsky01** [galaxyproject/galaxy#16019]
**Describe the bug**

Tab and push notifications for job completion no longer function with the new history. When enabled by users, the notifications create a count or a red galaxy logo on the tab bar, as well as a push notification whenever a job completes and a user is not focused on the Galaxy tab. With the new history, the history does not update until it is in focus, meaning the notifications will never be sent

**Galaxy Version and/or server at which you observed the bug**
US, EU

**To Reproduce**
1. Turn on job notifications from the User preferences
2. Run job
3. Exit to new tab before job completes
4. Wait for job to complete

**Expected behavior**
Tab should either update to show "Galaxy (x)" with "x" being the number of jobs completed, or the galaxy logo should turn red to demonstrate an error, and a push notification should be sent for job completion either way

**Additional context**
Discussion needed for how to fix, because the refresh only on focus was intended to decrease server load

**Comment by mvdbeek** [comment]
Yes, we stop polling the history for changes when it isn't active. It's very expensive to poll for history datasets (we have never actually polled for jobs) when the tab isn't active. We should probably remove the notification, it only seems marginally useful.

**Comment by dannon** [comment]
Two things:

- Instead of stopping, dramatically back off polling when in background.  1-2m or so, instead of stopped.
- Add client preference configurability for notifications
